# Issue 32909: Fix deprecation warnings with matplotlib 3.5

archive/issues_032672.json:
```json
{
    "body": "`get_path_in_displaycoord` was made private in matplotlib 3.5 and its public counterpart was deprecated, which causes a few test failures.\n\nSince this is only used inside a custom hack to circumvent some undesired upstream behavior, it seems to me that using private API here is not that bad. Ideally it should be ported to use public API, but it doesn't seem straightforward.\n\nUpdating matplotlib is deferred to a separate ticket, as it needs new dependencies (fonttools) and causes another test failure due to an upstream bug:\n\nhttps://github.com/matplotlib/matplotlib/issues/21683\n\n**CC:**  @kiwifb\n\n**Branch/Commit:** [a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)\n\n**Upstream:** Fixed upstream, in a later stable release.\n\n**Reviewer:** Fran\u00e7ois Bissey\n\n**Author:** Antonio Rojas\n\nIssue created by migration from https://trac.sagemath.org/ticket/32909\n\n",
    "closed_at": "2021-12-12T15:09:16Z",
    "created_at": "2021-11-20T08:59:05Z",
    "labels": [
        "component: packages: standard",
        "critical",
        "bug"
    ],
    "milestone": "https://github.com/sagemath/sage/milestones/sage-9.5",
    "title": "Fix deprecation warnings with matplotlib 3.5",
    "type": "issue",
    "url": "https://github.com/sagemath/sage/issues/32909",
    "user": "https://github.com/antonio-rojas"
}
```
`get_path_in_displaycoord` was made private in matplotlib 3.5 and its public counterpart was deprecated, which causes a few test failures.

Since this is only used inside a custom hack to circumvent some undesired upstream behavior, it seems to me that using private API here is not that bad. Ideally it should be ported to use public API, but it doesn't seem straightforward.

Updating matplotlib is deferred to a separate ticket, as it needs new dependencies (fonttools) and causes another test failure due to an upstream bug:

https://github.com/matplotlib/matplotlib/issues/21683

**CC:**  @kiwifb

**Branch/Commit:** [a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)

**Upstream:** Fixed upstream, in a later stable release.

**Reviewer:** Fran√ßois Bissey

**Author:** Antonio Rojas

Issue created by migration from https://trac.sagemath.org/ticket/32909





---

archive/issue_comments_534263.json:
```json
{
    "body": "**Branch:** [u/arojas/fix_deprecation_warnings_with_matplotlib_3_5](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/fix_deprecation_warnings_with_matplotlib_3_5)",
    "created_at": "2021-11-20T09:00:30Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534263",
    "user": "https://github.com/antonio-rojas"
}
```

**Branch:** [u/arojas/fix_deprecation_warnings_with_matplotlib_3_5](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/fix_deprecation_warnings_with_matplotlib_3_5)



---

archive/issue_comments_534264.json:
```json
{
    "body": "<a id='comment:2'></a>\n**New commits:**\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7\">a5127dc</a></td><td><code>get_path_in_displaycoord was made private in matplotlib 3.5</code></td></tr></table>\n",
    "created_at": "2021-11-20T09:06:50Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534264",
    "user": "https://github.com/antonio-rojas"
}
```

<a id='comment:2'></a>
**New commits:**
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7">a5127dc</a></td><td><code>get_path_in_displaycoord was made private in matplotlib 3.5</code></td></tr></table>




---

archive/issue_events_297513.json:
```json
{
    "actor": "https://github.com/antonio-rojas",
    "created_at": "2021-11-20T09:06:50Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "component: packages: standard",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297513"
}
```



---

archive/issue_comments_534265.json:
```json
{
    "body": "**Commit:** [a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)",
    "created_at": "2021-11-20T09:06:50Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534265",
    "user": "https://github.com/antonio-rojas"
}
```

**Commit:** [a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)



---

archive/issue_comments_534266.json:
```json
{
    "body": "**Author:** Antonio Rojas",
    "created_at": "2021-11-20T09:06:50Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534266",
    "user": "https://github.com/antonio-rojas"
}
```

**Author:** Antonio Rojas



---

archive/issue_comments_534267.json:
```json
{
    "body": "**Description changed:**\n``````diff\n--- \n+++ \n@@ -1 +1,7 @@\n+`get_path_in_displaycoord` was made private in matplotlib 3.5 and its public counterpart was deprecated, which causes a few test failures.\n \n+Since this is only used inside a custom hack to circumvent some undesired upstream behavior, it seems to me that using private API here is not that bad. Ideally it should be ported to use public API, but it doesn't seem straightforward.\n+\n+Updating matplotlib is deferred to a separate ticket, as it needs new dependencies (fonttools) and causes another test failure due to an upstream bug:\n+\n+https://github.com/matplotlib/matplotlib/issues/21683\n``````\n",
    "created_at": "2021-11-20T09:06:50Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534267",
    "user": "https://github.com/antonio-rojas"
}
```

**Description changed:**
``````diff
--- 
+++ 
@@ -1 +1,7 @@
+`get_path_in_displaycoord` was made private in matplotlib 3.5 and its public counterpart was deprecated, which causes a few test failures.
 
+Since this is only used inside a custom hack to circumvent some undesired upstream behavior, it seems to me that using private API here is not that bad. Ideally it should be ported to use public API, but it doesn't seem straightforward.
+
+Updating matplotlib is deferred to a separate ticket, as it needs new dependencies (fonttools) and causes another test failure due to an upstream bug:
+
+https://github.com/matplotlib/matplotlib/issues/21683
``````




---

archive/issue_events_297514.json:
```json
{
    "actor": "https://github.com/antonio-rojas",
    "created_at": "2021-11-20T09:06:50Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297514"
}
```



---

archive/issue_events_297515.json:
```json
{
    "actor": "https://github.com/antonio-rojas",
    "created_at": "2021-11-20T09:09:12Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "bug",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297515"
}
```



---

archive/issue_comments_534268.json:
```json
{
    "body": "**Upstream:** Fixed upstream, in a later stable release.",
    "created_at": "2021-11-23T17:36:43Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534268",
    "user": "https://github.com/dimpase"
}
```

**Upstream:** Fixed upstream, in a later stable release.



---

archive/issue_comments_534269.json:
```json
{
    "body": "<a id='comment:4'></a>\nthe upstream issue has been fixed, it seems. At least one can get a patch.",
    "created_at": "2021-11-23T17:36:43Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534269",
    "user": "https://github.com/dimpase"
}
```

<a id='comment:4'></a>
the upstream issue has been fixed, it seems. At least one can get a patch.



---

archive/issue_comments_534270.json:
```json
{
    "body": "<a id='comment:5'></a>\nI'd still like to leave the upgrade to a separate ticket to get this in asap and make sure it makes it to 9.5 (thus saving distros some downstream patching)",
    "created_at": "2021-11-26T21:15:07Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534270",
    "user": "https://github.com/antonio-rojas"
}
```

<a id='comment:5'></a>
I'd still like to leave the upgrade to a separate ticket to get this in asap and make sure it makes it to 9.5 (thus saving distros some downstream patching)



---

archive/issue_comments_534271.json:
```json
{
    "body": "<a id='comment:6'></a>\nTotally agree, we had that situation too late in the release cycle for sage 9.4 with matplotlib 3.4. Let's not miss the boat this time :)\n\nLGTM",
    "created_at": "2021-11-26T22:36:39Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534271",
    "user": "https://github.com/kiwifb"
}
```

<a id='comment:6'></a>
Totally agree, we had that situation too late in the release cycle for sage 9.4 with matplotlib 3.4. Let's not miss the boat this time :)

LGTM



---

archive/issue_comments_534272.json:
```json
{
    "body": "**Reviewer:** Fran\u00e7ois Bissey",
    "created_at": "2021-11-26T22:36:39Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534272",
    "user": "https://github.com/kiwifb"
}
```

**Reviewer:** Fran√ßois Bissey



---

archive/issue_events_297516.json:
```json
{
    "actor": "https://github.com/kiwifb",
    "created_at": "2021-11-26T22:36:39Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297516"
}
```



---

archive/issue_events_297517.json:
```json
{
    "actor": "https://github.com/kiwifb",
    "created_at": "2021-11-26T22:36:39Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297517"
}
```



---

archive/issue_events_297518.json:
```json
{
    "actor": "https://github.com/slel",
    "created_at": "2021-11-26T23:04:09Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "critical",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297518"
}
```



---

archive/issue_events_297519.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2021-12-12T15:09:16Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297519"
}
```



---

archive/issue_events_297520.json:
```json
{
    "actor": "https://github.com/vbraun",
    "commit_id": "8b176aaa0a4061fbe638edaf4c45b1b25de00c87",
    "created_at": "2021-12-12T15:09:16Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32909#event-297520"
}
```



---

archive/issue_comments_534273.json:
```json
{
    "body": "**Changing branch** from \"[u/arojas/fix_deprecation_warnings_with_matplotlib_3_5](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/fix_deprecation_warnings_with_matplotlib_3_5)\" to \"[a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)\".",
    "created_at": "2021-12-12T15:09:16Z",
    "issue": "https://github.com/sagemath/sage/issues/32909",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32909#issuecomment-534273",
    "user": "https://github.com/vbraun"
}
```

**Changing branch** from "[u/arojas/fix_deprecation_warnings_with_matplotlib_3_5](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/fix_deprecation_warnings_with_matplotlib_3_5)" to "[a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7](https://github.com/sagemath/sagetrac-mirror/commit/a5127dc56fdf5c2e82f6bc781cfe78dbd04e97b7)".
